Test/ContentPipeline/TestProcessorContext.cs

```diff
@@ -49,10 +49,12 @@ public override OpaqueDataDictionary Parameters
             get { throw new NotImplementedException(); }
         }
 
+#if !XNA
         public override ContentIdentity SourceIdentity
```

Comment:
We should remove the SourceIdentity from ContentProcessorContext and leave it as part of the PipelineProcessorContext. The user can get the source/filename with `((PipelineProcessorContext)context).SourceIdentity`.

In general whenever we extent on the XNA API this way we should do so with adding new interfaces.
like so: `public ContentIdentity IFilebasedContentProcessorContext.SourceIdentity { get; }`

Reply:
i guess i could see the point in moving it to the other class, while leaving it public to avoid changing the usage pattern/visibility. But hiding api extensions designed to make the framework easier to use behind otherwise unexposed interfaces defeats the purpose.

Reply:
We really don't need to make simple extensions like this one that difficult and ugly to use. It only affects someone if they are using it and shared code between MG and XNA...  Like we are for unit tests.

I don't think this needs to change here.

However we could go back and make 'SourceIdentity' virtual with a default implementation instead of abstract.